homeassistant/components/switch/amcrest.py

```diff
@@ -0,0 +1,96 @@
+"""
+This component supports turning on/off motion detection on Amcrest IP cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.amcrest/
+"""
+
+import logging
+import voluptuous as vol
+from requests.exceptions import HTTPError, ConnectTimeout
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_USERNAME, CONF_PASSWORD,
+    CONF_PORT, STATE_UNKNOWN, STATE_OFF, STATE_ON)
+from homeassistant.helpers.entity import ToggleEntity
+
+REQUIREMENTS = ['amcrest==1.2.0']
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Amcrest'
+DEFAULT_PORT = 80
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for an Amcrest IP Camera."""
+    from amcrest import AmcrestCamera
+
+    camera = AmcrestCamera(
+        config.get(CONF_HOST), config.get(CONF_PORT),
+        config.get(CONF_USERNAME), config.get(CONF_PASSWORD)).camera
+
+    try:
```

Comment:
@dparker18  could you add the persistent_notification feature as https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/amcrest.py#L59-L70